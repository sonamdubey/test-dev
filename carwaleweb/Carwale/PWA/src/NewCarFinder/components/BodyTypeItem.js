import React from 'react'
import PropTypes from 'prop-types'

const propTypes = {
	// body type data
	item: PropTypes.object,
	// body type count
	showCount: PropTypes.bool,
	// body type icon
	iconType: PropTypes.string
}

const defaultProps = {
	showCount: true,
	iconType: 'color'
}

class BodyTypeItem extends React.PureComponent {
	constructor(props) {
		super(props)
	}

	render(){
		const {
			item,
			showCount,
			iconType
		} = this.props

		let icon

		if (iconType === 'line') {
			icon = item.lineIcon
		}
		else {
			icon = item.icon
		}

		return (
			<div>
				<div className="item__icon-content">
					<img src={icon} alt={item.name} className="item__icon" />
				</div>
				<p className="item__heading">
					{item.name}
					{showCount ? " (" + item.carCount + ")" : ""}
				</p>
			</div>
		)
	}
}

BodyTypeItem.propTypes = propTypes
BodyTypeItem.defaultProps = defaultProps

export default BodyTypeItem
