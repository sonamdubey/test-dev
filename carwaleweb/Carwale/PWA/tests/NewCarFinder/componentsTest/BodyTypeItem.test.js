import React from 'react'
import BodyTypeItem from '../../../src/NewCarFinder/components/BodyTypeItem'


describe("Tests for BodyTypeItem component", () => {
    let props = {
        item : {
            "id": 6,
            "name": "SUV/MUV",
            "carCount": 8,
            "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nam elementum.",
            "icon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv_clr.svg",
            "lineIcon": "https://imgd.aeplcdn.com//0x0/cw/body/svg/suv.svg",
            "isSelected": true,
            "isRecommended": true
        }
    }

    let enzymeWrapper

    const Wrapper = () => {
        if(!enzymeWrapper){
            enzymeWrapper = shallow(<BodyTypeItem { ...props } />)
        }
        return enzymeWrapper
    }
    beforeEach(() => {
        enzymeWrapper = undefined
    })
    describe("when item is not selected", () => {
        it("should match snapshot", () => {
            props.item.isSelected = false
            const wrapper = Wrapper()
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("when item is selected", () => {
        it("should match snapshot", () => {
            props.item.isSelected = true
            const wrapper = Wrapper()
            expect(wrapper).toMatchSnapshot()
        })
    })
})
