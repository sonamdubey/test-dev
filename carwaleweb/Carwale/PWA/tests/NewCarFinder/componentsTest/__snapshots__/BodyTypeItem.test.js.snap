// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for BodyTypeItem component when item is not selected should match snapshot 1`] = `
<div>
  <div
    className="item__icon-content"
  >
    <img
      alt="SUV/MUV"
      className="item__icon"
      src="https://imgd.aeplcdn.com//0x0/cw/body/svg/suv_clr.svg"
    />
  </div>
  <p
    className="item__heading"
  >
    SUV/MUV
     (8)
  </p>
</div>
`;

exports[`Tests for BodyTypeItem component when item is selected should match snapshot 1`] = `
<div>
  <div
    className="item__icon-content"
  >
    <img
      alt="SUV/MUV"
      className="item__icon"
      src="https://imgd.aeplcdn.com//0x0/cw/body/svg/suv_clr.svg"
    />
  </div>
  <p
    className="item__heading"
  >
    SUV/MUV
     (8)
  </p>
</div>
`;
